File: angular-front-end/src/app/model/shopping-cart.ts
import { CartItem } from './cart-item';

export class ShoppingCart {
  id: string | null = null;
  userId: number = 0;
  total: number = 0;
  status: boolean = false;
  createdAt: Date = new Date();
  cartItems: CartItem[] = [];

  constructor() {
  }
}

File: catalogue-service/src/main/java/org/nc/entity/ShoppingCart.java
package org.nc.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Date;

@Document
@Data @AllArgsConstructor @NoArgsConstructor
public class ShoppingCart {
    @Id
    private String id;
    private Long userId;
    private int total;
    private boolean status;
    private Date createdAt;
    private boolean delivered;

    private Collection<CartItem> cartItems = new ArrayList<>();

}
